databroker/eventsource/archiver.py

```diff
@@ -10,45 +10,113 @@
 import pytz
 import uuid
 
+import numpy as np
+
 import requests
 import pandas as pd
 
+from ..utils import ALL
 
 class ArchiverEventSource(object):
-    def __init__(self, url, timezone, pvs):
+    
+    def __init__(self, config):
         """
-        DataBroker plugin
-
-        Parameters
-        ----------
-        url : string
-            e.g., 'http://host:port/'
-        timezone : string
-            e.g., 'US/Eastern'
-        pvs : dict
-            a dict mapping human-friendly names to PVs
+        Shim class to turn the EPICS Archiver Appliance into EventSource
         """
+
+        if all (k in config for k in ['name', 'url']):
+            pass
+        else:
+            raise TypeError("config {} does not include one of required"
+                            " keys (name, url).".format(config))
+        
+        self._name = config['name']
+        
+        url =  config['url']
         if not url.endswith('/'):
             url += '/'
         self.url = url
         self.archiver_addr = self.url + "retrieval/data/getData.json"
+
+        if 'timezone' in config:
+            timezone = config['timezone']
+        else:
+            timezone = 'US/Eastern'
         self.tz = pytz.timezone(timezone)
+
+        if 'pvs' in config:
```

Comment:
Is there a compelling use case for making `pvs` optional? Should we just require it?

Reply:
pvs is optional since their choice depends on data analysis and can be updated via the add_pvs method.